### ISSUE TITLE
False positive "X packages not found in AUR"

### ISSUE THREAD

**Issue by yan12125** [jjk-jacky/kalu#41]
```
$ kalu -m   
2 updates available (D: 838.55 KiB; N: 13.00 KiB)
- <b>lib32-harfbuzz</b> 1.1.3-1 > <b>1.2.3-1</b> (D: 145.31 KiB; N: -44.00 KiB)
- <b>openssh</b> 7.1p2-1 > <b>7.2p1-1</b> (D: 693.25 KiB; N: 57.00 KiB)
64 packages not found in AUR
- <b>abootimg-git</b> (0.6.r6.g1ebeb39-1)
- <b>acroread</b> (9.5.5-5)
- <b>acroread-fonts</b> (9.1.0-1)
- <b>android-apktool</b> (2.0.3-1)
- <b>android-google-repository</b> (r24-1)
- <b>android-ndk</b> (r10e-1)
- <b>android-platform</b> (23_r02-1)
- <b>android-sdk</b> (24.4.1-1)
- <b>android-sdk-build-tools</b> (r23.0.2-1)
- <b>android-sdk-platform-tools</b> (r23.1.0-2)
- <b>android-sources</b> (23_r01-1)
- <b>android-support</b> (r23.2-1)
- <b>android-support-repository</b> (r25-1)
- <b>apache-flex-sdk</b> (4.15.0-1)
- <b>camlidl</b> (1.05-5)
- <b>clearlooks-phenix-gtk-theme</b> (6.0.3-2)
- <b>dex2jar</b> (2.0-1)
- <b>dhcptest-git</b> (0.5.r8.ge67802e-1)
- <b>dmg2img</b> (1.6.5-4)
- <b>dropbox</b> (3.14.7-1)
- <b>ext4fuse-git</b> (v0.1.r55.gba01a66-1)
- <b>fb-adb-git</b> (20160216.r334.37ba40c-1)
- <b>ffdec</b> (8.0.1-1)
- <b>flashplayer-standalone-debug</b> (11.2.202.569-1)
- <b>freshplayerplugin</b> (0.3.4-1)
- <b>gapi-ocaml</b> (0.2.8-1)
- <b>google-chrome</b> (48.0.2564.116-1)
- <b>google-drive-ocamlfuse</b> (0.5.22-1)
- <b>google-talkplugin</b> (5.41.0.0-3)
- <b>heimdall-nogui-git</b> (1.4.1.r25.ga59a290-1)
- <b>jd-gui</b> (1.4.0-3)
- <b>jdk6</b> (6u45-4)
- <b>js-beautify</b> (1.5.10-2)
- <b>kalu</b> (4.0.2-1)
- <b>libav-git-no-libs</b> (v12_dev0.r2434.gf6ccee9-1)
- <b>libtinfo-5</b> (5-10)
- <b>logcat-color</b> (0.6.0-1)
- <b>make-3.81</b> (3.81-1)
- <b>nbuexplorer-bin</b> (3.3-1)
- <b>ncurses5-compat-libs</b> (6.0-2)
- <b>nodejs-jsonlint</b> (1.6.2-2)
- <b>ocaml-cryptokit</b> (1.10-1)
- <b>ocaml-curl</b> (0.5.3-5)
- <b>ocaml-easy-format</b> (1.0.2-1)
- <b>ocaml-extlib</b> (1.7.0-2)
- <b>ocaml-sqlite3</b> (4.0.2-2)
- <b>ocaml-xmlm</b> (1.2.0-1)
- <b>ocaml-yojson</b> (1.1.8-2)
- <b>ocamlfuse</b> (2.7.1-5)
- <b>ocamlnet</b> (4.0.4-2)
- <b>package-query-git</b> (1.7.158.gbdaacb1-1)
- <b>pcmanx-gtk2</b> (1.3-1)
- <b>python-pysvn</b> (1.8.0-1)
- <b>python26</b> (2.6.9-6)
- <b>python32</b> (3.2.6-1)
- <b>rabcdasm-git</b> (1.18.r0.g68f6fa2-1)
- <b>simg2img-git</b> (r17339.1e2382a-1)
- <b>socat2-git</b> (2.0.0.b6.120.g2dee720-1)
- <b>spim-svn</b> (689-1)
- <b>tortoisehg</b> (3.7.1-1)
- <b>vim-youcompleteme-git</b> (1644.f25e1c9-1)
- <b>virtualbox-ext-oracle</b> (5.0.14-1)
- <b>xfce4-volumed-pulse</b> (0.2.0-5)
- <b>yaourt-git</b> (1.7.119.gbce2a0d-1)
AUR: 2 watched packages updated
- <b>package-query</b> 1.7-2 > <b>-</b>
- <b>yaourt</b> 1.7-1 > <b>-</b>
```
Obviously most of these packages are in AUR. Seems there's something changed at AUR:
```
$ curl "https://aur.archlinux.org/rpc.php?type=multiinfo&arg\[\]=ocaml-xmlm&arg\[\]=ocaml-yojson" 
{"version":1,"type":"multiinfo","resultcount":0,"results":[]}

$ curl "https://aur.archlinux.org/rpc.php?v=5&type=multiinfo&arg\[\]=ocaml-xmlm&arg\[\]=ocaml-yojson"
{"version":5,"type":"multiinfo","resultcount":2,"results":[{"ID":206384,"Name":"ocaml-xmlm","PackageBaseID":45414,"PackageBase":"ocaml-xmlm","Version":"1.2.0-1","Description":"OCaml xml manipulation module","URL":"http:\/\/erratique.ch\/software\/xmlm\/","NumVotes":22,"Popularity":1.521243,"OutOfDate":null,"Maintainer":"J5lx","FirstSubmitted":1295067763,"LastModified":1435934289,"URLPath":"\/cgit\/aur.git\/snapshot\/ocaml-xmlm.tar.gz","Depends":["ocaml"],"MakeDepends":["opam"],"License":["BSD"],"Keywords":[]},{"ID":201807,"Name":"ocaml-yojson","PackageBaseID":45086,"PackageBase":"ocaml-yojson","Version":"1.1.8-2","Description":"An optimized parsing and printing library for JSON","URL":"http:\/\/mjambon.com\/yojson.html","NumVotes":19,"Popularity":1.519158,"OutOfDate":null,"Maintainer":"leonardder","FirstSubmitted":1294198025,"LastModified":1435068017,"URLPath":"\/cgit\/aur.git\/snapshot\/ocaml-yojson.tar.gz","Depends":["ocaml-easy-format"],"MakeDepends":["ocaml-biniou","ocaml-findlib","cppo"],"License":["BSD"],"Keywords":[]}]}
```
The following fix works for me, though I'm not sure whether it should be fixed in AUR or kalu.
```Diff
diff --git a/configure.ac b/configure.ac
index aa45c9a..72e7c44 100644
--- a/configure.ac
+++ b/configure.ac
@@ -41,7 +41,7 @@ AC_ARG_WITH([news-rss-url],
 # Options for AUR URL
 AC_ARG_WITH([url-aur-prefix],
         AC_HELP_STRING([--with-url-aur-prefix=URL], [set the prefix for the AUR URL]),
-        [AUR_URL_PREFIX=$withval], [AUR_URL_PREFIX="https://aur.archlinux.org/rpc.php?type=multiinfo"])
+        [AUR_URL_PREFIX=$withval], [AUR_URL_PREFIX="https://aur.archlinux.org/rpc.php?v=5&type=multiinfo"])
 AC_ARG_WITH([url-aur-prefix-pkg],
        AC_HELP_STRING([--with-url-aur-prefix-pkg=PREFIX],
                [set the prefix before each package for the AUR URL]),
```

**Comment by jjk-jacky** [comment]
Yes, new version of the AUR which requires to tweak the URL used
in kalu indeed.

That patch looks good, I don't have time right now but I'll commit later
-- Thanks!

**Comment by yan12125** [comment]
I guess this ticket can be closed as both kalu and kalu-git works fine now?

**Comment by jjk-jacky** [comment]
Sure; Though they've been "fixed" via using the configure option on
their respective PKGBUILDs, so it's unrelated to kalu as upstream
project.

In that regard, the fix was done in commit 2c7f0756, and I believe the
issue will be closed automatically, but only when the commit will be
merged in master (as per github behavior).
